src/git-source-provider.ts

```diff
@@ -109,8 +110,17 @@ export async function getSource(settings: IGitSourceSettings): Promise<void> {
 
     // Fetch
     core.startGroup('Fetching the repository')
-    const refSpec = refHelper.getRefSpec(settings.ref, settings.commit)
-    await git.fetch(settings.fetchDepth, refSpec)
+    if (settings.fetchDepth <= 0) {
+      let refSpec = refHelper.getRefSpecForAllHistory(settings.ref)
+      await git.fetch(refSpec)
```

Comment:
todo: test scenario where a new commit was pushed (not force push). Going to need some additional changes to this PR to checkout the correct commit